datahub-web-react/src/app/entity/shared/siblingUtils.ts

```diff
@@ -41,6 +42,31 @@ const combineMerge = (target, source, options) => {
     return destination;
 };
 
+const mergeTags = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'tag.urn');
+};
+
+const mergeTerms = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'term.urn');
+};
+
+const mergeAssertions = (destinationArray, sourceArray, _options) => {
+    return unionBy(destinationArray, sourceArray, 'urn');
+};
+
+function getArrayMergeFunction(key) {
+    switch (key) {
+        case 'tags':
+            return mergeTags;
+        case 'terms':
+            return mergeTerms;
+        case 'assertions':
+            return mergeAssertions;
+        default:
+            return undefined;
```

Comment:
Will this cause any runtime issues when we add new arrays to our objects?

Reply:
no sir - originally we weren't even setting `arrayMerge` which effectively we were setting `arrayMerge: undefined`. So it doesn't break anything, just defaults to how we were doing it before this fix. If we add more arrays to our objects that we need to remove duplicates to, we will need to add another custom array merge function like we have for these 3 fields so far.